src/blockstore/memory.rs

```diff
@@ -46,11 +46,18 @@ impl RootedBlockStore for MemoryBlockStore {
     }
 }
 
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
same here, not sure this needs to be arch gaurded

Reply:
these kinds of tests dont get run when we run wasm-pack test, and a lot of the tests i guarded this way use crates that wasm doesnt need or cant use, like tokio. IMO makes more sense to create a separate ticket for turning tests like these which bear shared responsibility into #[wasm_bindgen_test]s depending on architecture